tests/rpc-tests/rpc-compat.js

```diff
@@ -0,0 +1,199 @@
+  }
+  console.log("Balance of node account: ", balance)
+
+  return [machine, web3, nodeAccount]
+  
+}
+
+function fetchTestFiles(dir) {
+  try {
+    let testData = []
+
+    fs.readdirSync(dir).forEach((file) => {
+      const filePath = path.join(dir, file)
+      const stat = fs.statSync(filePath)
+
+      if (stat.isDirectory()) {
+        testData = testData.concat(fetchTestFiles(filePath))
+      } else {
+        const data = fs.readFileSync(filePath, 'utf8')
+        const jsonData = JSON.parse(data)
+        testData.push(jsonData)
+      }
+    })
+
+    return testData
+  } catch (error) {
+    console.error('Error occured while fetching test rpc data: ', error)
+  }
+}
+
+export async function rpcCompat() {
+  try {
+    let [machine, web3, sender] = await init()
+
+    // Getter rpc calls
+    const getTestData = fetchTestFiles('../../tests/rpc-tests/testdata/getters')
+    const axiosInstance = axios.create({
+      baseURL: `http://${machine}:8545`,
+    })
+
+    let response
+    
+    for (let i = 0; i < getTestData.length; i++) {
+      console.log('📍 Executing: ', getTestData[i].req.method)
+      response = await axiosInstance.post('/', getTestData[i].req)
+      assert.deepStrictEqual(response.data, getTestData[i].res)
+    }  
+
+    // Sender rpc calls
+    const sendTestData = fetchTestFiles(
+      '../../tests/rpc-tests/testdata/senders'
+    )
+
+    const nonce = await web3.eth.getTransactionCount(sender)
+
+    // prepare sender address
+    for (let i = 0; i < sendTestData.length; i++) {
+      if (sendTestData[i].req.method === 'eth_sign') {
+        sendTestData[i].req.params[0] = sender
+      } else if (sendTestData[i].req.method === 'eth_sendTransaction' || sendTestData[i].req.method === 'eth_signTransaction') {
+        sendTestData[i].req.params[0].from = sender
+      } 
+      
+      // raw transaction to be used in eth_sendRawTranscation
+      if (sendTestData[i].req.method === 'eth_signTransaction') {
+        sendTestData[i].req.params[0].nonce = web3.utils.toHex(nonce)
+        response = await axiosInstance.post('/', sendTestData[i].req)
+        console.log(`signed tx: ${response.data.result.raw}`)
+      }
+    }
+
+    for (let i = 0; i < sendTestData.length; i++) {
+      if (sendTestData[i].req.method === 'eth_sendRawTransaction') {
+        sendTestData[i].req.params[0] = response.data.result.raw
+        break
+      }
+      
+    }
+
+    const getFilterChanges = {"method":"eth_getFilterChanges","params":[""],"id":1,"jsonrpc":"2.0"}
+    let filterResponse
+
+    const getFilterLogs = {"method":"eth_getFilterLogs","params":[""],"id":1,"jsonrpc":"2.0"}
+    let filterLogsResponse
+
+    // Finally, fire 'em
+    for (let i = 0; i < sendTestData.length; i++) {
+      console.log('📍 Executing: ', sendTestData[i].req.method)
+      response = await axiosInstance.post('/', sendTestData[i].req)
+      if (response.data.error !== undefined && response.data.error !== null) {
```

Comment:
As discussed, let's not only check for errors, but also - and especially - for data sanity/correctness